File: bench/programs/generics/generics.py
import time


class Wrapper:
    __slots__ = ("value",)

    def __init__(self, value):
        self.value = value

    def get(self):
        return self.value


def run():
    n = 10000000
    p = 1000000007
    acc = 0
    for i in range(n):
        w = Wrapper(i)
        acc = (acc + w.get() * 2) % p
    return acc


run()  # warmup
t0 = time.perf_counter_ns()
ans = run()
dt = time.perf_counter_ns() - t0
print(ans)
print(dt)

File: bench/programs/generics/generics.go
package main

import (
	"fmt"
	"time"
)

type Wrapper[T any] struct {
	value T
}

func (w Wrapper[T]) get() T { return w.value }

func run() int64 {
	var n, p int64 = 10000000, 1000000007
	var acc int64 = 0
	for i := int64(0); i < n; i++ {
		w := Wrapper[int64]{value: i}
		acc = (acc + w.get()*2) % p
	}
	return acc
}

func main() {
	run() // warmup
	t0 := time.Now()
	ans := run()
	dt := time.Since(t0).Nanoseconds()
	fmt.Println(ans)
	fmt.Println(dt)
}
